### PR TITLE
Remove invalid assert for TextHeightBehavior

### PR DESCRIPTION
**Pull request by GaryQian**
## Description

https://github.com/flutter/flutter/pull/48346#issuecomment-584839221 caught an extra assert that slipped in. Remove this.

`null` is a valid value for this and there should not be an assert here.

### CODE REVIEW

**Review by justinmc — APPROVED** [review]
LGTM, thanks for the quick fix!

File: packages/flutter/lib/src/painting/text_painter.dart [hunk 1]
```diff
@@ -346,7 +346,6 @@ class TextPainter {
   ui.TextHeightBehavior get textHeightBehavior => _textHeightBehavior;
   ui.TextHeightBehavior _textHeightBehavior;
   set textHeightBehavior(ui.TextHeightBehavior value) {
-    assert(value != null);
```
Comment (dnfield) [inline packages/flutter/lib/src/painting/text_painter.dart]: We should add a test for this so someone doesn't helpfull add it back...

Comment (dnfield) [reply packages/flutter/lib/src/painting/text_painter.dart]: (probably a very simple test with a link to the github issue for context)

**Review by dnfield — CHANGES_REQUESTED** [review]
Needs a test.

**Comment by GaryQian** [comment]
Done!

**Review by dnfield — APPROVED** [review]
LGTM

**Pull request merged**